funds-manager/funds-manager-server/src/custody_client/withdraw.rs

```diff
@@ -55,30 +55,36 @@ impl CustodyClient {
             .get_vault_account(source.vault_name())
             .await?
             .ok_or_else(|| FundsManagerError::Custom("Vault not found".to_string()))?;
-
-        let asset = self.get_wallet_for_ticker(&vault, symbol).ok_or_else(|| {
-            FundsManagerError::Custom(format!("Asset not found for symbol: {}", symbol))
+        let asset_id = self.get_asset_id_for_address(mint).await?.ok_or_else(|| {
+            FundsManagerError::Custom(format!("Asset not found for mint: {mint}"))
         })?;
 
         // Check if the available balance is sufficient
+        let available = vault
+            .assets
+            .iter()
+            .find(|a| a.id == asset_id)
+            .map(|acct| acct.available.clone())
+            .unwrap_or_default();
```

Comment:
are we sure we want to `unwrap_or_default` here? that would result in the "Insufficient balance" error whereas a clearer one would be something to the tune of `FundsManagerError::Custom("Asset not found in vault")`

Reply:
those are the same thing